### PR TITLE
Remove cheatnet dependency from sncast

### LINKED ISSUE #1419: Remove cheatnet dependency from sncast
https://github.com/foundry-rs/starknet-foundry/pull/1401/files#diff-3623497f34cf1bd0ededf31f95e511abb9ae5b8791f80cce7dd7e8e066031ab6R27-R28

may be moved to the `runtime` crate

### PR DESCRIPTION
**Pull request by abulenok**
Closes #1419 

## Introduced changes

- `build_block_context`, `build_transaction_context` and related constants moved to runtime/context
- `DictStateReader` moved to runtime/context

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Comment by abulenok** [comment]
This PR is a continuation of https://github.com/foundry-rs/starknet-foundry/pull/1401

**Review by MaksymilianDemitraszek — APPROVED** [review]
Looked briefly, I am ok to do this but please wait for another review

File: crates/runtime/src/context.rs
Comment (piotmag769) [inline crates/runtime/src/context.rs:140]: Stuff from there to the end of the file should be in a separate file I think, maybe `state.rs`?

**Review by piotmag769 — APPROVED** [review]

**Review by THenry14 — APPROVED** [review]

**Review by Arcticae — COMMENTED** [review]
> `build_block_context`, `build_transaction_context` and related constants moved to runtime/context

I am confused with this decision, isn't runtime supposed to be generic and not rely on the starknet context? Do we need to load these into memory always?

File: crates/runtime/src/context.rs
Comment (abulenok) [reply crates/runtime/src/context.rs]: improved

**Pull request merged**
